### PR TITLE
WRO-12382: Increased code coverage for ui/Slider, ui/Heading, and ui/Marquee

### PR DESCRIPTION
**Pull request by daniel-stoian-lgp**
### Checklist

* [x] I have read and understand the [contribution guide](http://enactjs.com/docs/developer-guide/contributing/)
* [ ] A [CHANGELOG entry](http://enactjs.com/docs/developer-guide/contributing/changelogs/) is included
* [ ] At least one test case is included for this feature or bug fix
* [x] Documentation was added or is not needed
* [ ] This is an API breaking change

### Issue Resolved / Feature Added
[//]: # (Describe the issue resolved or feature added by this pull request)
Increased code coverage for ui/Slider, ui/Heading and ui/Marquee by adding extra unit tests

### Resolution
[//]: # (Does the code work as intended?)
[//]: # (What is the impact of this change and *why* was it made?)

### Additional Considerations
[//]: # (How should the change be tested?)
[//]: # (Are there any outstanding questions?)
[//]: # (Were any side-effects caused by the change?)

### Links
[//]: # (Related issues, references)
WRO-12382

### Comments
Enact-DCO-1.0-Signed-off-by: Daniel Stoian [daniel.stoian@lgepartner.com](mailto:daniel.stoian@lgepartner.com)

### CODE REVIEW

File: packages/ui/Marquee/tests/Marquee-specs.js [hunk 1]
```diff
@@ -147,8 +166,60 @@ describe('Marquee', () => {
...
+		const marquee = screen.getByText(ltrText);
+
+		fireEvent.focus(marquee);
+
+		act(() => jest.advanceTimersByTime(100));
+
+		expect(marquee).toHaveStyle({'--ui-marquee-spacing': '50'});
+
+		// calling blur for code coverage purposes. onBlur does not trigger any visual changes in jsdom.
+		fireEvent.blur(marquee.parentElement.parentElement);
```
Comment (daniel-stoian-lgp) [inline packages/ui/Marquee/tests/Marquee-specs.js:206]: for the reviewer: the onBlur handler is attached to the parent of the MarqueeComponent 
image

**Review by adrian-cocoara-lgp — APPROVED** [review]
LGTM

File: packages/ui/Marquee/tests/Marquee-specs.js [hunk 2]
```diff
@@ -147,8 +166,60 @@ describe('Marquee', () => {
...
+				<Marquee marqueeOn="render" marqueeOnRenderDelay={-10} marqueeSpacing={80} />
+			</Marquee>
+		);
+
+		act(() => jest.advanceTimersByTime(100));
+
+		const expected = 1;
+		const actual = spy.mock.calls.length;
+
+		expect(actual).toBe(expected);
```
Comment (MikyungKim) [inline packages/ui/Marquee/tests/Marquee-specs.js:183]: How about
```
expect(spy).toHaveBeenCalled();
```
?

File: packages/ui/Marquee/tests/Marquee-specs.js (hunk 1 above)
Comment (MikyungKim) [inline packages/ui/Marquee/tests/Marquee-specs.js:206]: I think we could add a test 'should stop marquee on blur if `marqueeOn` is focus` for testing blur handler. WDYT?

Comment (MikyungKim) [reply packages/ui/Marquee/tests/Marquee-specs.js:206]: Hm.. As you said, it looks like not triggering any visual changes whatsoever.

Comment (MikyungKim) [inline packages/ui/Marquee/tests/Marquee-specs.js:206]: Let's just remove it for now. Even if we remove this line, the coverage rate is above 85%.

File: packages/ui/Marquee/tests/Marquee-specs.js [hunk 3]
```diff
@@ -147,8 +166,60 @@ describe('Marquee', () => {
...
+		const marquee = screen.getByText(ltrText);
+
+		fireEvent.mouseOver(marquee);
+
+		act(() => jest.advanceTimersByTime(100));
+
+		expect(marquee).toHaveStyle({'--ui-marquee-spacing': '50'});
+
+		// calling mouseLeave code coverage purposes. MouseLeave does not trigger any visual changes in jsdom.
+		fireEvent.mouseLeave(marquee);
```
Comment (MikyungKim) [inline packages/ui/Marquee/tests/Marquee-specs.js:220]: Let's just remove it for now. Even if we remove this line, the coverage rate is above 85%.

File: packages/ui/Marquee/tests/Marquee-specs.js [hunk 4]
```diff
@@ -325,3 +396,93 @@ describe('MarqueeBase', () => {
...
+
+		fireEvent.focus(marquee1);
+
+		act(() => jest.advanceTimersByTime(100));
+
+		expect(marquee1).toHaveStyle({'--ui-marquee-spacing': '50'});
+		expect(marquee2).toHaveStyle({'--ui-marquee-spacing': '50'});
+
+		// calling blur for code coverage purposes. onBlur does not trigger any visual changes in jsdom.
+		fireEvent.blur(marquee1);
```
Comment (MikyungKim) [inline packages/ui/Marquee/tests/Marquee-specs.js:454]: same here.

File: packages/ui/Marquee/tests/Marquee-specs.js [hunk 5]
```diff
@@ -325,3 +396,93 @@ describe('MarqueeBase', () => {
...
+
+		fireEvent.mouseOver(marquee1);
+
+		act(() => jest.advanceTimersByTime(100));
+
+		expect(marquee1).toHaveStyle({'--ui-marquee-spacing': '50'});
+		expect(marquee2).toHaveStyle({'--ui-marquee-spacing': '50'});
+
+		// calling mouseLeave code coverage purposes. MouseLeave does not trigger any visual changes in jsdom.
+		fireEvent.mouseLeave(marquee1);
```
Comment (MikyungKim) [inline packages/ui/Marquee/tests/Marquee-specs.js:485]: here too.

File: packages/ui/Slider/tests/Slider-specs.js [hunk 6]
```diff
@@ -1,19 +1,87 @@
-import {render} from '@testing-library/react';
+import '@testing-library/jest-dom';
+import {fireEvent, render, screen} from '@testing-library/react';
 
+import Knob from '../Knob';
+import ProgressBar from '../../ProgressBar';
```
Comment (MikyungKim) [inline packages/ui/Slider/tests/Slider-specs.js:5]:
```suggestion
import ProgressBar from '../../ProgressBar';

import Knob from '../Knob';
```

File: packages/ui/Slider/tests/Slider-specs.js [hunk 7]
```diff
@@ -1,19 +1,87 @@
...
-		render(<Slider progressBarComponent={ProgressBar} ref={ref} />);
+		render(<Slider progressBarComponent={CustomProgressBar} ref={ref} />);
 
 		const expected = 'DIV';
 		const actual = ref.mock.calls[0][0].nodeName;
 
 		expect(actual).toBe(expected);
 	});
+
+	test('should set \'ui-slider-proportion-end-knob\' to 0 when \'defaultValue\' is smaller than min value', () => {
```
Comment (MikyungKim) [inline packages/ui/Slider/tests/Slider-specs.js:23]: I don't see any `defaultValue` prop from ui/Slider nor ui/ProgressBar.
Could you explain why you put this prop through the below tests?

Comment (daniel-stoian-lgp) [reply packages/ui/Slider/tests/Slider-specs.js:23]: Changeable is applied on SliderBase
From Changeables's jsdoc: 
_If `value` is passed to `Changeable`, the HOC assumes that the value is managed elsewhere and it
will not update its internal state. To set an initial value, use `defaultValue` instead._

Comment (daniel-stoian-lgp) [reply packages/ui/Slider/tests/Slider-specs.js:23]: in Sandstone/Slider, the jsdoc is more detailed explained, but I don't see any difference regarding value <-> defaultValue logic

_By default, `Slider` maintains the state of its `value` property. Supply the `defaultValue`
property to control its initial value. If you wish to directly control updates to the
component, supply a value to `value` at creation time and update it in response to `onChange`
events._

File: packages/ui/Marquee/tests/Marquee-specs.js (hunk 1 above)
Comment (daniel-stoian-lgp) [reply packages/ui/Marquee/tests/Marquee-specs.js:206]: Thank you for the idea :)

File: packages/ui/Slider/tests/Slider-specs.js (hunk 7 above)
Comment (MikyungKim) [reply packages/ui/Slider/tests/Slider-specs.js:24]: I see. I think we need to improve our documentation regarding this. We will discuss this through the workshop. Thanks!

**Review by MikyungKim — APPROVED** [review]
:lgtm:

**Pull request merged**